src/adapters/framed_tcp.rs

```diff
@@ -24,14 +24,28 @@ const INPUT_BUFFER_SIZE: usize = u16::MAX as usize; // 2^16 - 1
 
 #[derive(Clone, Debug, Default)]
 pub struct FramedTcpConnectConfig {
+    keepalive: Option<TcpKeepalive>,
+}
+
+impl FramedTcpConnectConfig {
     /// Enables TCP keepalive settings on the socket.
-    pub keepalive: Option<TcpKeepalive>,
+    pub fn with_keepalive(mut self, keepalive: TcpKeepalive) -> Self {
```

Comment:
I would reexport `TcpKeepalive` type to avoid the user creating a dependency of `socket2` only for its usage. Something like:

```rust
pub use socket2::TcpKeepalive;
```

in the top of the file.